### ISSUE TITLE
Tab accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#821]
**Component**
`Tab` (existing)

**Describe the feature request**
- Align `Tab` with the accessibility behaviour defined in the Figma specs

**Acceptance criteria**
- Roles should be correctly read out by the AT
- Tab state should be correctly read out by the AT
- Keyboard support is included
- Minimally test in NVDA + Chrome (Windows), VO + Safari (macOS), Talkback + Chrome (Android)

**References**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=359-11853&p=f&m=dev)
[Storybook](https://designsystem.life.gov.sg/react/index.html?path=/docs/content-tab--docs)
[WAI ARIA APG](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/)
